self-core-impl/src/main/java/com/selfxdsd/core/GitlabIssue.java

```diff
@@ -0,0 +1,214 @@
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+ * OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.selfxdsd.core;
+
+import com.selfxdsd.api.*;
+import com.selfxdsd.api.storage.Storage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.json.JsonObject;
+import javax.json.JsonValue;
+import java.net.URI;
+
+/**
+ * An Issue in a Gitlab repository.
+ * <br/>
+ * <a href="https://docs.gitlab.com/ee/api/issues.html#single-project-issue">
+ *  See specification </a>
+ *  <br/>
+ *  For a merge request
+ *  <a href ="https://docs.gitlab.com/ee/api/merge_requests.html#get-single-mr">
+ *     See specification </a>
+ * @author criske
+ * @version $Id$
+ * @since 0.0.38
+ * @todo 719#60 Implement and test method `assign()` for
+ *  GitlabIssue by following GithubIssue as model.
+ * @todo 720#60 Implement and test method `unassign()` for
+ *  GitlabIssue by following GithubIssue as model.
+ * @todo 721#60 Implement and test method `close()` for
+ *  GitlabIssue by following GithubIssue as model.
+ * @todo 722#60 Implement and test method `reopen()` for
+ *  GitlabIssue by following GithubIssue as model.
+ * @todo 723#60 Start implementing GitlabComments by following GithubComments
+ *  as model. These will be used by GitlabIssue.
+ * @todo 724#60 Start implementing GitlabLabels by following GithubLabels
+ *  as model. These will be used by GitlabIssue.
+ */
+final class GitlabIssue implements Issue {
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(
+        GitlabIssue.class
+    );
+
+    /**
+     * Issue base uri.
+     */
+    private final URI issueUri;
+
+    /**
+     * Issue JSON as returned by Gitlab's API.
+     */
+    private final JsonObject json;
+
+    /**
+     * Self storage, in case we want to store something.
+     */
+    private final Storage storage;
+
+    /**
+     * Gitlab's JSON Resources.
+     */
+    private final JsonResources resources;
+
+    /**
+     * Ctor.
+     * @param issueUri Issues base URI.
+     * @param json Json Issue as returned by Gitlab's API.
+     * @param storage Storage.
+     * @param resources Gitlab's JSON Resources.
+     */
+    GitlabIssue(
+        final URI issueUri,
+        final JsonObject json,
+        final Storage storage,
+        final JsonResources resources
+    ) {
+        this.issueUri = issueUri;
+        this.json = json;
+        this.storage = storage;
+        this.resources = resources;
+    }
+
+    @Override
+    public String issueId() {
+        return String.valueOf(this.json.getInt("iid"));
```

Comment:
I've opted for `iid` (relative id to project) insted of `id` (global id) because Issues and Pull Requests are using `iid` in their url:
`GET /projects/:id/issues/:issue_iid`
`GET /projects/:id/merge_requests/:merge_request_iid`